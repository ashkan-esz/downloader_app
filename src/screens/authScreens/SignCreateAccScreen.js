import React from 'react';
import {View, StyleSheet} from 'react-native';
import {SignInCreateAcc} from "../../components/molecules";
import {ScreenLayout} from "../../components/layouts";


const SignCreateAccScreen = ({navigation}) => {
    return (
        <ScreenLayout>
            <View style={style.container}>
                <SignInCreateAcc
                    onSignIn={() => {
                        navigation.navigate('SignIn')
                    }}
                    onCreatAcc={() => {
                        navigation.navigate('SignUp');
                    }}
                />
            </View>
        </ScreenLayout>
    );
};

const style = StyleSheet.create({
    container: {
        position: 'absolute',
        bottom: '9%',
        width: '85%',
        maxWidth: 350,
    },
});


export default SignCreateAccScreen;
