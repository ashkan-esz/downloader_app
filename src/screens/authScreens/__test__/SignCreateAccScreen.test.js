import * as React from "react";
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-dom';
import SignCreateAccScreen from "../SignCreateAccScreen";

describe("SignUp or create account screen ", () => {

    test('renders correctly', () => {
        let {toJSON} = render(
            <SignCreateAccScreen/>
        );
        expect(toJSON()).toMatchSnapshot();
    });

    test('navigation to SignIn', ()=> {
        let navigate = jest.fn();
        let {getByText} = render(
            <SignCreateAccScreen navigation={{navigate}}/>
        )
        let signInButton = getByText('Sign In');
        expect(navigate).toHaveBeenCalledTimes(0);
        fireEvent.press(signInButton);
        expect(navigate).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledWith('SignIn');
    });

    test('navigation to SignUp', ()=> {
        let navigate = jest.fn();
        let {getByText} = render(
            <SignCreateAccScreen navigation={{navigate}}/>
        )
        let createAccButton = getByText('Create Account');
        expect(navigate).toHaveBeenCalledTimes(0);
        fireEvent.press(createAccButton);
        expect(navigate).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledWith('SignUp');
    });

});
