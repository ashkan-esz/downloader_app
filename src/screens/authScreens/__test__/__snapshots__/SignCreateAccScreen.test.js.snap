// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp or create account screen  renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1D1D27",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "bottom": "9%",
        "maxWidth": 350,
        "position": "absolute",
        "width": "85%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "borderRadius": 25,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#E15050",
                "borderColor": "#2089dc",
                "borderRadius": 25,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Sign In
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 25,
            "marginTop": 15,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2f2f3b",
                "borderColor": "#2089dc",
                "borderRadius": 25,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Create Account
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
